modules/worker-utils/src/lib/worker-farm/worker-body.ts

```diff
@@ -2,15 +2,24 @@ import type {WorkerMessageData, WorkerMessageType, WorkerMessagePayload} from '.
 import {getTransferList} from '../worker-utils/get-transfer-list';
 
 /** Vile hack to defeat over-zealous bundlers from stripping out the require */
-function getParentPort() {
+async function getParentPort() {
   // const isNode = globalThis.process;
   let parentPort;
   try {
     // prettier-ignore
     eval('globalThis.parentPort = require(\'worker_threads\').parentPort'); // eslint-disable-line no-eval
     parentPort = globalThis.parentPort;
-    // eslint-disable-next-line no-empty
-  } catch {}
+  } catch (e) {
+    if ((e as Error).message === 'require is not defined') {
```

Comment:
This looks brittle, such error messages tend to change between versions.

Reply:
Done